### ISSUE TITLE
DOCTYPE definition not possible in data.xml

### ISSUE THREAD

**Issue by dark-m0de** [speedata/publisher#355]
Some HTML entities that are used within xml in VSCode cause warnings, e.g.:
- `&lt;`, `&gt;` and some others work fine
- `&nbsp;`or `&euro;` create a warning message.

VSCode proposes to create a DOCTYPE definition, e.g. for `&nbsp;`:
```xml
<?xml version="1.0"?>
<!DOCTYPE data [
  <!ENTITY nbsp "entity-value">
]>
<data>
</data>
```

However, adding a doctype definition to the data.xml crashes the speedata publisher:
```
Loading data file "/Users/conrad/Repositories/creisch-dev/creisch-fachinfo-template/data.xml"
Error: [page 1] ...ch-fachinfo-template/speedata-publisher/sw/lua/luxor.lua:69: bad argument #1 to 'gsub' (string expected, got nil)
stack traceback:
        ...-template/speedata-publisher/sw/lua/publisher/spinit.lua:57: in function 'call'
        ...-template/speedata-publisher/sw/lua/publisher/spinit.lua:408: in function 'main_loop'
        ...-template/speedata-publisher/sw/lua/publisher/spinit.lua:417: in main chunk
        [C]: in function 'require'
        [\directlua]:1: in main chunk
Stop processing data
1 errors occurred
```

May I kindly ask to review luxor.lua?

BR, Conrad

P.S.: I try my best to create a merge request in the future.

**Comment by dark-m0de** [comment]
P.P.S.: `&#160;` works as a workaround for `&nbsp;` to avoid warnings.

**Comment by pgundlach** [comment]
fixed in 4.7.7
